omnigan/trainer.py

```diff
@@ -1073,20 +1189,148 @@ def get_d_loss(self, multi_domain_batch, verbose=0):
                     if self.opts.gen.m.use_advent:
                         if verbose > 0:
                             print("Now training the ADVENT discriminator!")
-                        fake_mask = self.G.decoders["m"](z)
-                        fake_complementary_mask = 1 - fake_mask
-                        prob = torch.cat([fake_mask, fake_complementary_mask], dim=1)
-                        prob = prob.detach()
 
-                        loss_main = self.losses["D"]["advent"](
-                            prob.to(self.device),
-                            self.domain_labels[batch_domain],
-                            self.D["m"]["Advent"],
-                        )
+                        if "m2" in self.opts.tasks:
+                            # --------ADVENT LOSS---------------
```

Comment:
Add `CoBlock` to your vscode extension and use cmd+shift+k to make nice comment blocks instead of those atrocious imbalanced hybrid headers

Reply:
:p